### ISSUE TITLE
ParsingError: Optional Content PDF - with some PDF files uploaded

### ISSUE THREAD

**Issue by MatteoVella** [boazsegev/combine_pdf#230]
After uploading some PDF files, during the process of combining two pdfs, the process crashes and returns the following error:

> _CombinePDF::ParsingError: Optional Content PDF files aren't supported and their pages cannot be safely extracted._

I attach below one of the files that returned the error:
[pdf_test.pdf](https://github.com/boazsegev/combine_pdf/files/12656870/pdf_test.pdf)

While other PDF files like the following completed the process without any errors:
[pdf_test_2.pdf](https://github.com/boazsegev/combine_pdf/files/12656882/pdf_test_2.pdf)

What could the error be related to?

**Comment by MatteoVella** [comment]
Hi @boazsegev, can you help me, please?

**Comment by MatteoVella** [comment]
Hi @boazsegev, any news from this issue?

Let me know, thanks

**Comment by boazsegev** [comment]
Hi @MatteoVella 

Did you try allowing optional content (see [Known Limitations](https://github.com/boazsegev/combine_pdf#known-limitations))?

```ruby
filename = 'pdf_test.pdf'
CombinePDF.load(filename, allow_optional_content: true).
```

For me it works fine once that option is enabled.

**Comment by magnum** [comment]
thank you @boazsegev 
we're using both methods
```CombinePDF.load```
to load file from a local path
and 
```CombinePDF.parse```
to load a file via http
can we use `allow_optional_content: true` with both methods?

**Comment by boazsegev** [comment]
Yes, the extra options are passed through to the `parse` method.

Essentially `load` is just `parse` with an `IO.binread`.
